core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -139,9 +139,11 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
 		if(uri.endsWith("openapi.json")) {
 			uri = uri.substring(0, uri.lastIndexOf("/"));
 			ApiDispatchConfig apiConfig = dispatcher.findConfigForUri(uri);
-			JsonObject jsonSchema = dispatcher.generateOpenApiJsonSchema(apiConfig);
-			returnJson(response, 200, jsonSchema);
-			return;
+			if(apiConfig != null) {
+				JsonObject jsonSchema = dispatcher.generateOpenApiJsonSchema(apiConfig);
+				returnJson(response, 200, jsonSchema);
+				return;
+			}
 		}
```

Comment:
please add an error response in case no openapi.json can be returned because no matching config was found (apiConfig==null)

Reply:
error response is in line 161